apps/client/src/features/rundown/event-block/composite/EventBlockChip.tsx

```diff
@@ -0,0 +1,85 @@
+import { useMemo } from 'react';
+import { Tooltip } from '@chakra-ui/react';
+import type { MaybeNumber } from 'ontime-types';
+import { dayInMs, isPlaybackActive, MILLIS_PER_MINUTE, MILLIS_PER_SECOND } from 'ontime-utils';
+
+import { usePlayback, useTimelineStatus } from '../../../../common/hooks/useSocket';
+import { cx } from '../../../../common/utils/styleUtils';
+import { formatDuration } from '../../../../common/utils/time';
+import { tooltipDelayFast } from '../../../../ontimeConfig';
+
+import style from './EventBlockChip.module.scss';
+
+interface EventBlockChipProps {
+  id: string;
+  timeStart: number;
+  isPast: boolean;
+  isLoaded: boolean;
+  className: string;
+  accumulatedGap: MaybeNumber;
+  isNext: boolean;
+  isLinked: boolean;
+  isNextDay: boolean;
+}
+
+export default function EventBlockChip(props: EventBlockChipProps) {
+  const { timeStart, isPast, isLoaded, className, accumulatedGap, isNext, isLinked, isNextDay } = props;
+  const { playback } = usePlayback();
+
+  if (isLoaded) {
+    return null;
+  }
+
+  const playbackActive = isPlaybackActive(playback);
+
+  if (!playbackActive || isPast) {
+    return null; //TODO: Event report will go here
+  }
+
+  if (playbackActive) {
+    return (
+      <EventUntil
+        className={className}
+        timeStart={timeStart + (isNextDay ? dayInMs : 0)}
+        accumulatedGap={accumulatedGap}
+        isNextAndLinked={isNext && isLinked}
+      />
+    );
+  }
+
+  return null;
+}
+
+interface EventUntilProps {
+  className: string;
+  timeStart: number;
+  accumulatedGap: MaybeNumber;
+  isNextAndLinked: boolean;
+}
+
+function EventUntil(props: EventUntilProps) {
+  const { timeStart, className, accumulatedGap, isNextAndLinked } = props;
+  const { clock, offset } = useTimelineStatus();
+
+  const [timeUntilString, isDue] = useMemo(() => {
+    const gap = accumulatedGap ?? 0;
+
+    const consumedOffset = isNextAndLinked ? offset : Math.min(offset + gap, 0);
+    const offsetTimestart = timeStart - consumedOffset;
+    const timeUntil = offsetTimestart - clock;
+    const isDue = timeUntil < MILLIS_PER_SECOND;
+    return [isDue ? 'DUE' : `${formatDuration(Math.abs(timeUntil), timeUntil > 2 * MILLIS_PER_MINUTE)}`, isDue];
+  }, [accumulatedGap, clock, isNextAndLinked, offset, timeStart]);
+
+  if (offset === null) {
+    return null; //TODO: change offset to a maybe number
```

Comment:
question: can offset be null?

Reply:
not yet, but I would like to make it so
it makes some things easier and I think it makes sense to say that nothing is playing so we don't have a valid offset value
and in this specific case is solves the issue of the playback playback triggerring a render befor offset has a valid value

Reply:
I think we should only write code for a null offset when that is a valid case

As for making offset nullable, as I remember, the reason to make it always a number was to avoid a bunch of checks at runtime. but if you think this is better we can change it in a separate PR